src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -2015,13 +2015,30 @@ public override BoundNode VisitReturnStatement(BoundReturnStatement node)
             }
         }
 
-        private BetterResult BetterConversionTarget(TypeSymbol type1, TypeSymbol type2, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        private BetterResult BetterConversionTarget(
+            TypeSymbol type1,
+            TypeSymbol type2,
+            ref HashSet<DiagnosticInfo> useSiteDiagnostics,
+            int betterConversionTargetRecursionLimit = 100)
```

Comment:
Entries to these functions from outside the recursion (i.e. other places in the compiler) should not have to be aware of this parameter. That is why I prefer this pattern.